action/protocol/staking/protocol.go

```diff
@@ -54,8 +62,18 @@ func (p *Protocol) Handle(ctx context.Context, act action.Action, sm protocol.St
 		return p.handleDepositToStake(ctx, act, sm)
 	case *action.Restake:
 		return p.handleRestake(ctx, act, sm)
+	default:
+		return nil, errors.New("invalid staking message")
 	}
-	return nil, nil
+
+	r := action.Receipt{}
+
+	// store the candidate list back
```

Comment:
I thought we are going to store delegates one by one as we discussed offline yesterday.

Reply:
right, this need to be change per latest discussion